File: frontend/lib/models/ocr_result_model.dart
class OCRResultModel {
  final int id;
  final int documentId;
  final String? detectedType;
  final double detectedAmount;
  final String? detectedParty;
  final String? detectedCategory;
  final DateTime? detectedDate;
  final double confidence;
  final DateTime? createdAt;

  OCRResultModel({
    required this.id,
    required this.documentId,
    required this.detectedType,
    required this.detectedAmount,
    required this.detectedParty,
    required this.detectedCategory,
    required this.detectedDate,
    required this.confidence,
    required this.createdAt,
  });

  factory OCRResultModel.fromJson(Map<String, dynamic> json) {

    final ocr = Map<String, dynamic>.from(
      json['ocr_result'] ?? {},
    );

    return OCRResultModel(
      id: ocr['id'],
      documentId: ocr['document_id'],
      detectedType: ocr['detected_type'],
      detectedAmount: (ocr['detected_amount'] as num?)?.toDouble() ?? 0.0,
      detectedParty: ocr['detected_party'],
      detectedCategory: ocr['detected_category'],
      detectedDate: ocr['detected_date'] != null
          ? DateTime.tryParse(ocr['detected_date'].toString())
          : null,
      confidence: (ocr['confidence'] as num?)?.toDouble() ?? 0.0,
      createdAt: ocr['created_at'] != null
          ? DateTime.tryParse(ocr['created_at'].toString())
          : null,
    );
  }
}

File: backend/app/models/ocr_result_model.py
from sqlalchemy import Column, Integer, Float, String, DateTime, ForeignKey
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from database import Base

class OCRResult(Base):
    __tablename__ = "ocr_results"

    id = Column(Integer, primary_key=True)
    business_id = Column(Integer, ForeignKey("businesses.id"), nullable=False, index=True)
    document_id = Column(Integer, ForeignKey("documents.id"), nullable=False, index=True)
    detected_type = Column(String)   # sale | expense
    detected_amount = Column(Float, nullable=True)
    detected_party = Column(String, nullable=True)
    detected_category = Column(String, nullable=True)
    detected_date = Column(String, nullable=True)
    raw_text = Column(String)
    confidence = Column(Float)
    created_at = Column(DateTime, server_default=func.now())

    business = relationship("Business", back_populates="ocr_results")
